### PR TITLE
Make nested form rendering more robust and more flexible.

### PR DESCRIPTION
**Pull request by apfelbox**
| Q             | A
| ------------- | --------------------------------------------------------------------- |
| BC breaks?    | no                                                                |
| New feature?  | no                    |
| Improvement?  | yes     |
| Bug fix?      | no                                                                |
| Deprecations? | no     |
| Docs PR       | **missing**                                   |

### CODE REVIEW

File: src/Resources/views/form/theme.html.twig [hunk 1]
```diff
@@ -75,7 +75,7 @@
         {{- form_rest(form) -}}
     {%- else -%}
         {# with label -> show as nested form (but only if it is not a choice #}
-        {%- if choices is not defined -%}
+        {%- if (is_nested_form ?? true) -%}
```
Comment (keichinger) [inline src/Resources/views/form/theme.html.twig]: Thinking this entire form nesting process a step further: I've often had the requirements to either make it very obvious that a nested form is indeed nested (e.g. indent it and/or show a border) or make it not so obvious and make it appear like it's a regular form element.
In the cases where I had to pretend a nested form element isn't nested, I had to manually render the entire form, which is not only repetitive but also somewhat error prone.

What I'm saying: Do we want to add support (probably via a form extension) to configure the nesting behaviour?

Comment (apfelbox) [reply src/Resources/views/form/theme.html.twig]: Yeah, just override the block for your widget and set this variable to either `false` (inline the form) or leave it as `true` (to show it nested), like this:

```twig
{%- block hyperlink_row -%}
    {%- set is_nested_form = false -%}
    {{- block("form_row") -}}
{%- endblock hyperlink_row -%}
```

Your question is the whole point of this PR, actually.

Comment (apfelbox) [reply src/Resources/views/form/theme.html.twig]: You should even be able to set this variable in `YourFormType::buildView()`, actually.

Comment (apfelbox) [reply src/Resources/views/form/theme.html.twig]: Just confirmed, you can also do:

```php
    /**
     * @inheritDoc
     */
    public function buildView (FormView $view, FormInterface $form, array $options)
    {
        $view->vars["is_nested_form"] = false;
    }
```
Both work, either in the template or your form type. It's a matter of taste, whether this rather belongs to the view (template) or the form itself (form type).

Comment (keichinger) [reply src/Resources/views/form/theme.html.twig]: This is perfect! Thank you ❤️ 

I wasn't actually sure what this CSS class is supposed to do as we have no styling in this bundle :)

**Pull request merged**